### ISSUE TITLE
Moodle 3.2 compatibility

### ISSUE THREAD

**Issue by danielneis** [fredcy/moodle-tool_datatables#3]
Hello,

i am trying to use this plugin with version 3.2 and getting the following error:

    TypeError: col is undefined

It occurs on the following line:

https://github.com/fredcy/moodle-tool_datatables/blob/master/amd/src/jquery.dataTables.js#L6441

I am using the simple version:

    $PAGE->requires->js_call_amd('tool_datatables/init', 'init', array('.progress_right table', array("select" => false, "paginate" => false)

I tried calling it before and after `OUTPUT->header` with the same report.

Any ideas?

**Comment by fredcy** [comment]
I'll take a look at this soon.

**Comment by fredcy** [comment]
I'm using this plugin with Moodle version 3.2.1 and it's working fine for me. 

When you do see the error? Does it appear right away when you visit the page? Does it occur after some user event (mouse click, keyboard event into search box, etc)?

Does the test page work OK for you, the one found at Site administration > Server > Datatables > Datatables test?

**Comment by danielneis** [comment]
Yes, the test page seems to be ok, here is the screenshot. There are no errors on webconsole.

![screenshot from 2017-03-08 19 32 43](https://cloud.githubusercontent.com/assets/17608/23727305/47cc5a28-0436-11e7-90d8-fb73a700e1f9.png)

Do I need the aria-controls attribute on the th? Also, the sorting class?

Are there know problems when I have more then one tr with th ? In my case i have three rows of headers and I would like to use the last one to sort columns.

**Comment by fredcy** [comment]
I just built a bare Moodle 3.2.1 system from scratch and then installed the Datatables plugin v0.3. It works fine. I was concerned that some new dependency might have caused problems, but it looks fine to me.

I don't know anything about "aria-controls" or what you mean by "sorting class".

However, the multiple `TH` rows could be a problem for the third party datatables code itself (independent of being used in the Moodle context).  I haven't tried that. Docs suggest that it should work though: https://datatables.net/examples/basic_init/complex_header.html

This moodle plugin embeds a copy of version 1.10.9 of DataTables. I see that the latest available version of DataTables is now 1.10.13.

**Comment by danielneis** [comment]
The version available here on github is ready for deploy? Just need to do a git clone or need to run grunt or something else?

**Comment by fredcy** [comment]
The version here at github should be usable directly via git clone as I have been committing generated files to git and so everything needed is available. (I don't recall just why I did it that way. I think it's so I can install the plugin directly on a system without build tools like grunt.)
